sdk/identity/identity/src/client/identityClient.ts

```diff
@@ -188,11 +200,60 @@ export class IdentityClient extends ServiceClient implements INetworkModule {
     }
   }
 
+  // Here is a custom layer that allows us to abort requests that go through MSAL,
+  // since MSAL doesn't allow us to pass options all the way through.
+
+  private abortControllers?: Record<string, AbortController[]>;
+
+  generateAbortSignal(correlationId?: string): AbortSignalLike {
+    if (!this.abortControllers) {
+      this.abortControllers = {};
+    }
+
+    const controller = new AbortController();
```

Comment:
do these ever get cleaned up? Or do we hold onto them forever?

Reply:
we clean them with shift, but the main correlationIds don't get cleaned in this design. I can clean them if they're empty. I'll push that later today.